target/linux/mediatek/filogic/base-files/lib/preinit/10_fix_eth_mac.sh

```diff
@@ -4,21 +4,29 @@ preinit_set_mac_address() {
 	case $(board_name) in
 	acer,predator-w6|\
 	acer,predator-w6d)
-		$(mmc_get_mac_ascii u-boot-env WANMAC)
-		$(mmc_get_mac_ascii u-boot-env LANMAC)
-		ip link set dev lan1 address "$lan_mac"
-		ip link set dev lan2 address "$lan_mac"
-		ip link set dev lan3 address "$lan_mac"
-		ip link set dev game address "$lan_mac"
-		ip link set dev eth1 address "$wan_mac"
+		wan_mac=$(mmc_get_mac_ascii u-boot-env WANMAC)
+		lan_mac=$(mmc_get_mac_ascii u-boot-env LANMAC)
+		if [ -n "$lan_mac" ] ; then
```

Comment:
I think there is no need to check the MAC address. If they are empty, system can throw a scripts exit warning.

Reply:
What do you mean exactly with "scripts exit warning"? The whole point of this patch is to get rid of these ugly error messages:

    /etc/preinit: line 7: c0:98:79:xx:yy:zz: not found
    /etc/preinit: line 8: c0:98:79:xx:yy:zz: not found
    ip: "" is invalid lladdr
    ip: "" is invalid lladdr
    ip: "" is invalid lladdr
    ip: "" is invalid lladdr
    ip: "" is invalid lladdr

Reply:
Missing MAC address is an emergency error. We must fix it instead of ignoring it.

Reply:
AFAICT this just means that the uboot environment does not contain settings for mac addresses and this is totally fine as the hardware defaults are used. Why do you think this is a problem at all?

Reply:
> AFAICT this just means that the uboot environment does not contain settings for mac addresses

Yes.

> and this is totally fine as the hardware defaults are used. Why do you think this is a problem at all?

No. These embedded devices don't have default MAC address, they just use random values. In some countries, this may be illegal.